File: metadata/mill-metadata-core/src/main/java/io/qpointz/mill/metadata/repository/FacetTypeRepository.java
package io.qpointz.mill.metadata.repository;

import io.qpointz.mill.metadata.domain.FacetTypeDescriptor;

import java.util.Collection;
import java.util.Optional;

public interface FacetTypeRepository {
    void save(FacetTypeDescriptor descriptor);
    Optional<FacetTypeDescriptor> findByTypeKey(String typeKey);
    Collection<FacetTypeDescriptor> findAll();
    void deleteByTypeKey(String typeKey);
    boolean existsByTypeKey(String typeKey);
}

File: metadata/mill-metadata-core/src/main/kotlin/io/qpointz/mill/metadata/repository/FacetTypeRepository.kt
package io.qpointz.mill.metadata.repository

import io.qpointz.mill.metadata.domain.FacetType

/** Runtime facet type rows (`metadata_facet_type`) — SPEC §6.3. */
interface FacetTypeRepository {
    fun findByKey(typeKey: String): FacetType?

    fun findAll(): List<FacetType>

    fun findDefined(): List<FacetType>

    fun findObserved(): List<FacetType>

    fun save(facetType: FacetType): FacetType

    fun delete(typeKey: String)
}
